clconf/template_test.go

```diff
@@ -0,0 +1,390 @@
+	path := filepath.Join(temp, subPath)
+	err := MkdirAllNoUmask(path, perms)
+	if err != nil {
+		t.Fatalf("Error making temp sub dir %q: %v", path, err)
+	}
+	stat, err := os.Stat(path)
+	if err != nil {
+		t.Fatalf("Error stating folder %q after creation: %v", path, err)
+	}
+	if stat.Mode()&0777 != perms {
+		t.Fatalf("Created folder %q does not have proper permissions after creation [%o != %o]",
+			path, stat.Mode()&0777, perms)
+	}
+}
+
+func writeTestFile(t *testing.T, temp string, subPath string, perms os.FileMode) {
+	path := filepath.Join(temp, subPath)
+	content := []byte("{{ getv \"/foo\" }}")
+	err := ioutil.WriteFile(path, content, perms)
+	if err != nil {
+		t.Fatalf("Error making temp file %q: %v", path, err)
+	}
+	os.Chmod(path, perms)
+	stat, err := os.Stat(path)
+	if err != nil {
+		t.Fatalf("Error stating file %q after creation: %v", path, err)
+	}
+	if stat.Mode() != perms {
+		t.Fatalf("Created file %q does not have proper permissions after creation [%o != %o]",
+			path, stat.Mode(), perms)
+	}
+}
+
+func buildTestFolder(t *testing.T) string {
+	extension := ".clconf"
+	temp, err := ioutil.TempDir("", "")
+	if err != nil {
+		t.Fatalf("Error making temp folder %v", err)
+	}
+	makeTestSubfolder(t, temp, "subdir1/subsubdir1", 0775)
+	makeTestSubfolder(t, temp, "subdir1/subsubdir2", 0775)
+	makeTestSubfolder(t, temp, "subdir2", 0775)
+	makeTestSubfolder(t, temp, "emptydir", 0775)
+
+	writeTestFile(t, temp, "yes_basedir.html"+extension, 0646)
+	writeTestFile(t, temp, "yes_basedir1.html"+extension, 0640)
+	writeTestFile(t, temp, "yes_basedir2.html"+extension, 0640)
+	writeTestFile(t, temp, "no_basedir.html", 0641)
+	writeTestFile(t, temp, "no_basedir1.html", 0640)
+	writeTestFile(t, temp, "subdir1/yes_subdir1.sh"+extension, 0770)
+	writeTestFile(t, temp, "subdir1/no_subdir1.sh", 0770)
+	writeTestFile(t, temp, "subdir1/subsubdir1/yes_subdir1subsubdir1.sh"+extension, 0775)
+	writeTestFile(t, temp, "subdir1/subsubdir1/no_subdir1subsubdir1.sh", 0775)
+	writeTestFile(t, temp, "subdir1/subsubdir2/yes_subdir1subsubdir2.sh"+extension, 0777)
+	writeTestFile(t, temp, "subdir1/subsubdir2/no_subdir1subsubdir2.sh", 0777)
+	writeTestFile(t, temp, "subdir2/yes_subdir2.sh"+extension, 0777)
+	writeTestFile(t, temp, "subdir2/no_subdir2.sh", 0777)
+
+	return temp
+}
+
+func normalizePaths(paths []pathWithRelative) {
+	sort.SliceStable(paths, func(i, j int) bool {
+		return paths[i].full < paths[j].full || paths[i].rel < paths[j].rel
+	})
+}
+
+func testFindTemplates(t *testing.T, message string, extension string, subPath string, expectedFlat bool, expected []string) {
+	temp := buildTestFolder(t)
+	defer os.RemoveAll(temp)
+
+	paths, err := findTemplates(filepath.Join(temp, subPath), extension)
+	if err != nil {
+		t.Fatalf("Error running findTemplates (%s): %v", message, err)
+	}
+
+	if len(expected) == 0 {
+		if len(paths) != 0 {
+			t.Errorf("Paths wasn't empty when it was supposed to be: %v", paths)
```

Comment:
This error does not include `message`.

Reply:
Good catch, [fixed](https://github.com/pastdev/clconf/pull/11/commits/cff4dbfa560db473811dda5cfd43d26590448de6)